app/scripts/controllers/transactions.js

```diff
@@ -56,19 +67,22 @@ module.exports = class TransactionController extends EventEmitter {
         if (!account) return
         return account.balance
       },
-      publishTransaction: this.txProviderUtil.publishTransaction.bind(this.txProviderUtil),
-      getPendingTransactions: () => {
-        const network = this.getNetwork()
-        return this.getFilteredTxList({
-          status: 'submitted',
-          metamaskNetworkId: network,
-        })
+      publishTransaction: this.query.sendRawTransaction,
+      getPendingTransactions: this.txStateManager.getPendingTransactions.bind(this.txStateManager),
+      giveUpOnTransaction: (txId) => {
+        const msg = `Gave up submitting after 3500 blocks un-mined.`
+        this.setTxStatusFailed(txId, msg)
```

Comment:
is string correct or do we need to give it an err obj

Reply:
this dosent look like something i wrote must have come from merging master the other day

Reply:
but ill fix it

Reply:
✔️